sessionrecording/connect.go

```diff
@@ -7,19 +7,27 @@ package sessionrecording
 
 import (
 	"context"
+	"crypto/tls"
+	"encoding/json"
 	"errors"
 	"fmt"
 	"io"
 	"net"
 	"net/http"
 	"net/http/httptrace"
 	"net/netip"
+	"sync/atomic"
 	"time"
 
+	"golang.org/x/net/http2"
 	"tailscale.com/tailcfg"
+	"tailscale.com/util/httpm"
 	"tailscale.com/util/multierr"
 )
 
+// DialFunc is a function for dialing the recorder.
+type DialFunc func(context.Context, string, string) (net.Conn, error)
```

Comment:
`(_ context.Context, network, host string)` (or `ctx` instead of `_` if you prefer)